ivy/__init__.py

```diff
@@ -103,6 +103,8 @@ def __new__(cls, dev_str):
             ivy.utils.assertions.check_elem_in_list(dev_str[0:3], ["gpu", "tpu", "cpu"])
             if dev_str != "cpu":
                 # ivy.assertions.check_equal(dev_str[3], ":")
+                if len(dev_str) == 3:
```

Comment:
let's remove this, `torch` also allows not having to pass the `id` 😅
```
x = torch.tensor([1., 2., 3.], device="cuda")
```
So basically just like `gpu:100` falls back to `gpu:0`, `gpu` would also be read as `gpu:0`